### PR TITLE
[release] add golden notebook tests for dask/xgboost and modin/xgboost

### PR DESCRIPTION
**Pull request by matthewdeng**
## Why are these changes needed?

As part of the release test process, we will be adding a suite of "Golden Notebook" tests that test the integration of libraries and functionality that are often used together.

This PR adds 2 tests:
1. Dask + XGBoost Ray
2. Modin + XGBoost Ray

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/golden_notebook_tests/modin_xgboost_app_config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+base_image: "anyscale/ray:1.4-rc2"
```
Comment (richardliaw) [inline release/golden_notebook_tests/modin_xgboost_app_config.yaml:1]: should we use `latest` or something?

**Review by richardliaw — APPROVED** [review]
Workloads look fine to me. Will leave it to @krfricke and @amogkam to merge.

File: release/golden_notebook_tests/modin_xgboost_app_config.yaml (hunk 1 above)
Comment (matthewdeng) [reply release/golden_notebook_tests/modin_xgboost_app_config.yaml]: Updated to stable version 1.4.0

File: release/golden_notebook_tests/golden_notebook_tests.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+- name: dask_xgboost_test
+  owner:
+    mail: "matt@anyscale.com"
+    slack: "@ml-team"
```
Comment (krfricke) [inline release/golden_notebook_tests/golden_notebook_tests.yaml:4]:
```suggestion
    slack: "@team_ml"
```

File: release/golden_notebook_tests/golden_notebook_tests.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    compute_template: compute_tpl.yaml
+
+  run:
+    timeout: 900
+    script: python workloads/dask_xgboost_test.py
+
+- name: modin_xgboost_test
+  owner:
+    mail: "matt@anyscale.com"
+    slack: "@ml-team"
```
Comment (krfricke) [inline release/golden_notebook_tests/golden_notebook_tests.yaml:17]:
```suggestion
    slack: "@team_ml"
```

**Review by krfricke — APPROVED** [review]
This looks great!

**Pull request merged**